### ISSUE TITLE
tsserver: Format on type broken

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#5032]
See the animated git here: http://pan.baidu.com/s/1kTCNhdX

Format on type deletes valid code on the next line. This is reproducible using VSCode and tsserver from 1.6.2

**Comment by dbaeumer** [comment]
Since the bug is important for us I investigate. There are two issue:
- if the lineText has a '\t' the indentPosition is to be subtracted by the number of spaces in the text however later on in the replace edit only one character can be removed ('\t')
- if indentPosition is negative then the replace
```typescript
                                edits.push({
                                    span: ts.createTextSpanFromBounds(position, position - indentPosition),
                                    newText: ""
                                });
```
creates an edit that eats from position to the right. Hence it removes the 'va' in the animated gif.

**Comment by dbaeumer** [comment]
@mhegazy If you accept the PR can you please merge it to 1.6.x as well. This is really a bad experience in VSCode. I will even ship a patch version for this in this weeks drop.

**Comment by mhegazy** [comment]
@dbaeumer this part has not changed in a while? is this a regression? do you know when it was introduced?

**Comment by dbaeumer** [comment]
@mhegazy: no idea. I saw something comparable a while ago but then the formatter changed and I couldn't reproduce anymore. This one I could today on tsserver 1.6.2 with VSCode.

**Comment by dbaeumer** [comment]
@mhegazy I will work on test and the comments beginning of next week. Busy with VSCode this week.

**Comment by dbaeumer** [comment]
The PR also addressed #3681

**Comment by mhegazy** [comment]
no rush. feel free to update the PR at your convenience. thanks for fixing this!
